app/javascript/components/App.js

```diff
@@ -67,19 +66,84 @@ export default class App extends React.Component {
       })
   }
 
-  createNewBook = (form) => {
-    console.log(form);
+  createNewBook = (newBook) => {
+    console.log(newBook)
+    return fetch("/books", {
+      body: JSON.stringify(newBook),
+      headers: {
+        "Content-Type": "application/json"
+      },
+      method: "POST"
+    })
+      .then(response => {
+        if (response.status === 422) {
+          alert("Invalid Submission")
+        }
+        return response.json()
+      })
+      .then(payload => {
+        this.bookIndex()
+      })
+      .catch(errors => {
+        console.log("Create errors: ", errors)
+      })
+  }
+
+
+  updateBook = (book, id) => {
+    return fetch(`/books/${id}`, {
+      body: JSON.stringify(book),
+      headers: {
+        "Content-Type": "application/json"
+      },
+      method: "PATCH"
+    })
+      .then(response => {
+        if (response.status === 200) {
+          this.bookIndex()
+        }
+        return response
+      })
+      .catch(errors => {
+        console.log("edit errors:", errors)
+      })
   }
 
-  updateBook = (form, id) => {
-    console.log(form)
-    console.log(id)
+  deleteBook = (id) => {
+    return fetch(`/books/${id}`, {
+      headers: {
+        "Content-Type": "application/json"
+      },
+      method: "DELETE"
+    })
+      .then(response => {
+        if (response.status === 200) {
+          this.bookIndex()
+        }
+        return response.json()
+      })
+      .catch(errors => {
+        console.log("delete errors:", errors)
+      })
   }
 
   findItem = (arr, id) => {
     return arr.find(item => item.id === Number(id))
   }
 
+  findRentedBooks = (arr, id) => {
+    let rentedBooks = arr.filter(book => book.rentals.length > 0)
```

Comment:
I really love how you made 3 methods for doing this.  Turned out really nice.  I also like how you split this up into 2 calls.  It makes it easier to understand what is happening, step-by-step.  Just as an example of another approach, you could simplify this down to this:

```
let rentedBooks = arr.filter(book => book.rentals.length > 0 && book.rentals[0].user_id === id);
return rentedBooks;
```

You can make it even shorter by using an implicit return like so:

```
findRentedBooks = (arr, id) => arr.filter(book => book.rentals.length > 0 && book.rentals[0].user_id === id)
```

What you have is great, i'm just trying to show you some other ways of doing this.  👏

Reply:
Oh duh! Seeing the rework, that makes total sense. That reminded me that I can actually just make all the helpers one-liners so I did. Thanks!